lib/Backend/Lower.cpp

```diff
@@ -7616,6 +7616,7 @@ Lowerer::LowerUnaryHelperMem(IR::Instr *instr, IR::JnHelperMethod helperMethod,
     IR::Instr *instrPrev;
 
     instrPrev = LoadScriptContext(instr);
+    instr->m_func->SetHasThrow();
```

Comment:
> instr->m_func->SetHasThrow(); [](start = 3, length = 30)

LowerUnaryHelperMem is a generic API which could be called for non-throw opcodes as well. 
You should have this under case::Throw: